src/services/webdav/backend.rs

```diff
@@ -398,6 +355,42 @@ impl Accessor for WebdavBackend {
             _ => Err(parse_error(resp).await?),
         }
     }
+
+    async fn list(&self, path: &str, _: OpList) -> Result<(RpList, Self::Pager)> {
+        // XML body must start without a new line. Otherwise, the server will panic: `xmlParseChunk() failed`
+        let all_prop_xml_body = r#"<?xml version="1.0" encoding="utf-8" ?>
+            <D:propfind xmlns:D="DAV:">
+                <D:allprop/>
+            </D:propfind>
+        "#;
+
+        let async_body = AsyncBody::Bytes(bytes::Bytes::from(all_prop_xml_body));
+        let resp = self
+            .webdav_propfind(path, None, "application/xml".into(), async_body)
```

Comment:
Question: do we need to merge the function mode of these internal methods?
- propfind
- put
- mkcol

Reply:
The current design looks good! Each underlying API call in a separate function is easier to understand.

Reply:
Sounds good!